clientv3/watch.go

```diff
@@ -614,13 +603,42 @@ func (w *watchGrpcStream) serveSubstream(ws *watcherStream, resumec chan struct{
 				// shutdown from closeSubstream
 				return
 			}
-			// TODO pause channel if buffer gets too large
-			ws.buf = append(ws.buf, wr)
+
+			skipBuf := wr.Created && !ws.initReq.createdNotify
+			if wr.Created {
+				okToSend := false
+				select {
+				case ws.initReq.retc <- ws.outc:
+					okToSend = true
```

Comment:
`if ws.initReq.createdNotify { ws.outc <- *wr }` here instead of `okToSend` if getting rid of `skipBuf`?